packages/client/src/components/app/forms/InnerForm.svelte

```diff
@@ -408,16 +408,38 @@
     }
   }
 
-  const handleScrollToField = ({ field }) => {
-    const fieldId = get(getField(field)).fieldState.fieldId
-    const label = document.querySelector(`label[for="${fieldId}"]`)
-    document.getElementById(fieldId).focus({ preventScroll: true })
-    label.scrollIntoView({ behavior: "smooth" })
+  const handleScrollToField = ({ field, block }) => {
+    if (!field.fieldState) {
+      field = get(getField(field))
+    }
+    const fieldId = field.fieldState.fieldId
+    const fieldElement = document.getElementById(fieldId)
+    fieldElement.focus({ preventScroll: true })
+    if (block === "start") {
+      const label = document.querySelector(`label[for="${fieldId}"]`)
+      label.scrollIntoView({ behavior: "smooth" })
+    } else {
+      fieldElement.scrollIntoView({
+        behavior: "smooth",
+        block,
+      })
+    }
```

Comment:
As mentioned above, I think we should just use `nearest` for the `block` param here. We can ensure we always have a little bit of padding from the edge of the screen by using the `scrollMargin` propery.

e.g.
```js
fieldElement.style.scrollMargin = "100px"
fieldElement.scrollIntoView({
  behavior: "smooth",
  block: "nearest",
})
```

Reply:
The scrollMargin property is perfect! It makes sure the title of the form is also shown if the top most field is invalid.